boards/configurations/nrf52840dk/nrf52840dk-test-appid-tbf/src/main.rs

```diff
@@ -177,13 +176,15 @@ pub unsafe fn main() {
     nrf52840_peripherals.init();
     let base_peripherals = &nrf52840_peripherals.nrf52;
 
+    let processes = &*addr_of!(PROCESSES);
```

Comment:
This was already an issue before (as the kernel held onto an immutable reference of `PROCESSES`), but this change makes it very clear that we have mutable aliasing here. We must rethink how we keep and modify `PROCESSES` without creating unsoundness.